lib/internal/http2/compat.js

```diff
@@ -330,45 +306,94 @@ class Http2ServerResponse extends Stream {
     code |= 0;
     if (code >= 100 && code < 200)
       throw new RangeError('Informational status codes cannot be used');
-    if (code < 200 || code > 999)
+    if (code < 200 || code >= 600)
       throw new RangeError(`Invalid status code: ${code}`);
     state.statusCode = code;
   }
 
+  addTrailers(headers) {
+    for (var name of Object.keys(headers)) {
```

Comment:
I need to see if there are other areas where this needs to be changed, but please use a tradition `for (var n = 0; ...` style here. They out perform the `for-of` and `for-in` loops significantly.